### ISSUE TITLE
built-ins/TypedArray/prototype/byteOffset/resizable-array-buffer-auto.js and built-ins/TypedArray/prototype/byteOffset/BigInt/resizable-array-buffer-auto.js are wrong

### ISSUE THREAD

**Issue by marjakh** [tc39/test262#3206]
built-ins/TypedArray/prototype/byteOffset/BigInt/resizable-array-buffer-auto.js:

Line 39:

    ab.resize(BPE);
    expected = 0;

This should be the "at the boundary" case where the TA is *not* out of bounds.

built-ins/TypedArray/prototype/byteOffset/resizable-array-buffer-auto.js:

Line 37:

```
var expected = BPE;
  try {
    ab.resize(BPE);
    expected = 0;
  } catch (_) {}

  assert.sameValue(array.byteOffset, expected, "following shrink (on boundary)");
```

Ditto.

**Comment by marjakh** [comment]
The normative spec change for the "at the boundary" TAs should've been handled by this:

https://github.com/tc39/test262/pull/3163

But looks like the byteOffset tests aren't updated correctly; the expected value should be BPE, not 0.

cc @jugglinmike @rwaldron

**Comment by jugglinmike** [comment]
Thanks for reporting this! gh-3211 ought to fix things.

Scrutinizing gh-3163 further, it looks like I neglected a couple other BigInt tests (the ones for the `byteLength` and `length` accessor properties). The tests still happen to be technically correct, but their assertions incorrectly label the "on boundary" condition as "out of bounds," and they lack any verification of a true "out of bounds" condition. I've opened gh-3212 to correct that.
